docs/source/flow-project.rst

```diff
@@ -39,44 +39,104 @@ Executing this script on the command line will give us access to this project's
+
+.. code-block:: python
+
+    # init.py
+
+    import signac
+
+    project = signac.init_project('MyProject')
+    for i in range(10):
+        project.open_job({'a': i}).init()
+
+
+A very simple *operation*, which creates a file called ``hello.txt`` within a job's workspace directory, could be implemented like this:
+
+.. code-block:: python
+
+    # project.py
+
+    from flow import FlowProject
+
+    class MyProject(FlowProject):
+        pass
+
+    @MyProject.operation
+    def hello(job):
+        print('hello', job)
+        with job:
+            with open('hello.txt', 'w') as file:
+                file.write('world!\n')
+
+
+    if __name__ == '__main__':
+        MyProject().main()
+
+
+.. _conditions:
+
+Conditions
+==========
+
+Here the :py:meth:`~flow.FlowProject.operation` decorator function specifies that the ``hello`` operation function is part of our workflow. If we run ``python project.py run``, **signac-flow** will execute ``hello`` for all jobs in the project.
+
+However, we only want to execute ``hello`` if ``hello.txt`` does not yet exist in the job's workspace.
+To do this, we need to create a condition function named ``greeted`` that tells us if ``hello.txt`` already exists in the job workspace:
 
-We will reproduce the simple workflow introduced in the previous section by first copying both the ``greeted()`` condition function and the ``hello()`` *operation* function into the ``project.py`` module.
-We then use the :py:meth:`~flow.FlowProject.operation` and the :py:meth:`~flow.FlowProject.post` decorator functions to specify that the ``hello()`` operation function is part of our workflow and that it should only be executed if the ``greeted()`` condition is not met.
+
+.. code-block:: python
+
+    def greeted(job):
+        return job.isfile('hello.txt')
+
+To complete this component of the workflow, we use the :py:meth:`~flow.FlowProject.post` decorator function to specify that the ``hello`` operation function should only be executed if the ``greeted`` condition is *not* met.
+
+The entirety of the code is as follows:
 
 .. code-block:: python
 
     # project.py
     from flow import FlowProject
 
 
-    class Project(FlowProject):
+    class MyProject(FlowProject):
         pass
 
 
     def greeted(job):
         return job.isfile('hello.txt')
 
 
-    @Project.operation
-    @Project.post(greeted)
+    @MyProject.operation
+    @MyProject.post(greeted)
     def hello(job):
         with job:
             with open('hello.txt', 'w') as file:
                 file.write('world!\n')
 
 
     if __name__ == '__main__':
-        Project().main()
+        MyProject().main()
 
-We can define both *pre* and *post* conditions, which allow us to define arbitrary workflows as an acyclic graph.
+We can define both :py:meth:`~flow.FlowProject.pre` and :py:meth:`~flow.FlowProject.post` conditions, which allow us to define arbitrary workflows as an acyclic graph.
```

Comment:
We should link to a definition if we use "directed acyclic graph" as a math/CS term.
```suggestion
We can define both :py:meth:`~flow.FlowProject.pre` and :py:meth:`~flow.FlowProject.post` conditions, which allow us to define arbitrary workflows as a `directed acyclic graph <https://en.wikipedia.org/wiki/Directed_acyclic_graph>`__.
```